core/vm/lfvm/hash_cache.go

```diff
@@ -0,0 +1,246 @@
+package lfvm
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/crypto/sha3"
+)
+
+type hashCacheEntry32 struct {
+	key        [32]byte
+	hash       common.Hash
+	pred, succ *hashCacheEntry32
+}
+
+type hashCacheEntry64 struct {
+	key        [64]byte
+	hash       common.Hash
+	pred, succ *hashCacheEntry64
+}
+
+// HashCache is an LRU governed fixed-capacity cache for hash values.
```

Comment:
Perhaps you would like to refer to the SHA3 function.

Reply:
Done